pkg/ds/queue/queue_iter.go

```diff
@@ -0,0 +1,75 @@
+package queue
+
+// Iterator is not concurrent safe, use with care ^^.
+type Iterator[T any] struct {
+	queue *Queue[T]
+	curr  *Node[T]
+	start bool
+}
+
+func NewIter[T any](q *Queue[T]) *Iterator[T] {
+	return &Iterator[T]{
+		queue: q,
+		curr:  nil,
+	}
+}
+
+func (i *Iterator[T]) Val() (T, bool) {
+	var t T
+	if i.curr == nil {
+		return t, false
+	}
+
+	return i.curr.value, true
+}
+
+func (i *Iterator[T]) Next() bool {
+	if !i.start {
+		i.start = true
+		i.curr = i.queue.head
+		return true
+	}
+
+	if i.curr == nil || i.curr.next == nil {
+		return false
+	}
+
+	i.curr = i.curr.next
+	return true
+}
+
+func (i *Iterator[T]) RemoveCurrent() {
```

Comment:
`Iterator` should not modify `Queue` content, I think you should implement `Remove` method for `Queue`.

Reply:
remove iterator in https://github.com/KyberNetwork/tradinglib/pull/13/commits/c70340a02facf62735fa1cd710d2f2bd8d8338d1